agent/structs/structs.go

```diff
@@ -1217,6 +1218,11 @@ const (
 	// centralized configuration.
 	ServiceKindIngressGateway ServiceKind = "ingress-gateway"
 
+	// ServiceKindAPIGateway is an API Gateway for the Connect feature.
```

Comment:
Should we be using "Connect" anymore? Just say "service mesh"?

Reply:
This is me following the existing patterns, but I wouldn't mind updating the other comments nearby in this PR